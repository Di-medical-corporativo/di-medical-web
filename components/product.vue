<template>
    <b-card
      :img-src="photoThumb[0] || ''"
      :img-alt="product.title || ''"
      :sub-title="product.brand || ''"
      :title="product.title || ''"
      img-top
      tag="article"
      class="card mb-4"
    > 
      <b-card-text>
        {{shortenName || ''}}
      </b-card-text>

      <button class="card__button" @click="$router.push({ name: 'products-detail-id', params: { id: productTitle } })">Mas informacion</button>
      <span>
        <b-icon 
        :icon="product.stock? 'check2' : 'x'" 
        :variant="product.stock? 'success' : 'danger'"
        ></b-icon> 
          {{product.stock? 'En stock' : 'Agotado'}}
      </span>
    </b-card>  
</template>

<script>
import { defineComponent, computed } from "@nuxtjs/composition-api"


export default defineComponent({
  props: {
    product: {
      type: Object,
      default: () => {}
    }
  },
  setup({ product }) {
    const shortenName = computed(() => {
      return product.description.length > 150 ? product.description.substring(0, 140) + '...' : product.description
    })

    return {
      shortenName,
      photoThumb: computed(() => product.photos.filter(e => e.includes('jpg')) || []),
      date: computed(() =>{
        const date = new Date(product.date)
        return `${date.getDate()}-${date.getMonth() + 1}-${date.getFullYear()}`
      }),
      productTitle: computed(() => product.title.toLowerCase().split(' ').join('-'))
    }
  }

})
</script>

<style>

</style>