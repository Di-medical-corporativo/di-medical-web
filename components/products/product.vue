<template>
  <b-card
      title="Este es el nombre del producto"
      :img-src="product.photos[0]"
      :img-alt="photoThumb[0]"
      :sub-title="product.brand"
      img-top
      tag="article"
      class="card mb-4"
    > 
      <b-card-text>
        {{shortenName}}
      </b-card-text>

      <button class="card__button" @click="$router.push({ name: 'products-detail-id', params: { id: product.id } })">Mas informacion</button>
      <span>
        <b-icon 
        :icon="product.stock? 'check2' : 'x'" 
        :variant="product.stock? 'success' : 'danger'"
        ></b-icon> 
          {{product.stock? 'En stock' : 'Agotado'}}
      </span>
      <template #footer>
        <span>{{date}}</span>
      </template>
    </b-card> 
</template>

<script>
import { computed, defineComponent } from "@nuxtjs/composition-api";

export default defineComponent({
  props: {
    product: {
      type: Object,
      default: () => {}
    }
  },
  setup({ product }) {
    const shortenName = computed(() => {
      return product.description.length > 150 ? product.description.substring(0, 140) + '...' : product.description
    })

    return {
      shortenName,
      photoThumb: computed(() => product.photos.filter(e => e.includes('jpg'))),
      date: computed(() =>{
        const date = new Date(product.date)
        return `${date.getDate()}-${date.getMonth() + 1}-${date.getFullYear()}`
      })
    }
  }
})
</script>

<style>
</style>