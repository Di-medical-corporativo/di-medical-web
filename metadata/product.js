const productMeta = (title) => ({
    title: `${title} | Lo m치s selecto de las mejores marcas para el mercado hospitalario.`,
    meta: [
        { hid: 'description', name: 'description', content: 'Conoce nuestros productos, lo m치s selecto de las mejores marcas para el mercado hospitalario.' },
        { hid: 'author', name: 'author', content: 'Di-medical corporativo' },
        { hid: 'og-title', name: 'og:title', content: 'Productos | Lo m치s selecto de las mejores marcas para el mercado hospitalario.' },
        { hid: 'og-type', name: 'og:type', content: 'website' },
        { hid: 'og-url', name: 'og:url', content: `https://www.dimedicalcorporativo.mx/products` },
        { hid: 'og-description', name: 'og:description', content: 'Conoce nuestros productos, lo m치s selecto de las mejores marcas para el mercado hospitalario.' },
        { hid: 'og-image', name: 'og:image', content: 'https://firebasestorage.googleapis.com/v0/b/di-medical-del-sur.appspot.com/o/static%2FlogoCorporativo.png?alt=media&token=ca32a756-7656-4259-b5b7-921c11a0a3e8' }
      ]
})

export default productMeta