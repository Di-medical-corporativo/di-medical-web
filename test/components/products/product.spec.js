import product from '@/components/products/product.vue'
import { shallowMount } from '@vue/test-utils'

describe('product card', () => {
    const testProduct = {
        id: '123-abc',
        title: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.',
        photos: ['jpg', 'img2', 'img2',],
        description: 'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.',
        brand: 'Test brand for product',
        stock: true
    }

    test('should match snapshot', () => {
        const wrapper = shallowMount(product, {
            propsData: {
                product: testProduct
            }
        })
        expect(wrapper.html()).toMatchSnapshot()
    });

    test('should change stock icon depending on product stock propertie', () => {
        const wrapper = shallowMount(product, {
            propsData: {
                product: testProduct
            }
        })
        const stockIcon = wrapper.get('b-icon')
        expect(stockIcon.attributes('icon')).toBe('check2')
        expect(stockIcon.attributes('variant')).toBe('success')
    })

    test('should shorten name and title when length is greater than 120 and 40 respectively', () => {
        const wrapper = shallowMount(product, {
            propsData: {
                product: testProduct
            }
        })
        const shortenName = wrapper.vm.shortenName
        const shortenTitle = wrapper.vm.shortenTitle
        expect(shortenName.includes('...')).toBe(true)
        expect(shortenTitle.includes('...')).toBe(true)
    });
})
