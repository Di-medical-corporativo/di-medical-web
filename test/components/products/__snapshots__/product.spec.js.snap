// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product card should match snapshot 1`] = `
"<b-card img-src=\\"jpg\\" img-alt=\\"Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industrys standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book.\\" sub-title=\\"Test brand for product\\" title=\\"Lorem Ipsum is simpl...\\" img-top=\\"\\" tag=\\"article\\" class=\\"card mb-4\\">
  <b-card-text>
    Lorem Ipsum is simply dummy text of the printing a...
  </b-card-text> <button class=\\"card__button\\">
    Mas informacion
  </button> <span><b-icon icon=\\"check2\\" variant=\\"success\\"></b-icon>
    En stock
  </span>
</b-card>"
`;
